local.go

```diff
@@ -129,38 +129,42 @@ func minimize(settings *Settings, method Method, funcInfo *FunctionInfo, stats *
 	}
 
 	for {
+		if funcInfo.IsStatuser {
+			// Check the function status before evaluating.
+			status, err = funcs.status.Status()
+			if err != nil || status != NotTerminated {
+				return
+			}
+		}
+
+		// Perform evalType evaluation of the function at xNext and store the
+		// result in location.
+		evaluate(funcs, funcInfo, evalType, xNext, location, stats)
+		// Update the stats and optLoc (provided a new optimum has been found).
+		update(location, optLoc, stats, iterType, startTime)
+		// Get the convergence status before recording the new location.
+		status = checkConvergence(optLoc, iterType, stats, settings)
+
 		if settings.Recorder != nil {
 			err = settings.Recorder.Record(location, evalType, iterType, stats)
 			if err != nil {
-				status = Failure
```

Comment:
I guess that's reasonable. I was hesitant because I was pondering the implications of seeing NotTerminated as a status externally, but I suppose it's fine.